learning_resources/etl/sloan.py

```diff
@@ -0,0 +1,215 @@
+
+OFFERED_BY_CODE = OfferedBy.see.name
+
+
+def _get_access_token():
+    """
+    Get an access token for edx
+
+    Args:
+        config (OpenEdxConfiguration): configuration for the openedx backend
+
+    Returns:
+        str: the access token
+    """
+    payload = {
+        "grant_type": "client_credentials",
+        "client_id": settings.SEE_API_CLIENT_ID,
+        "client_secret": settings.SEE_API_CLIENT_SECRET,
+        "token_type": "jwt",
+    }
+    response = requests.post(  # noqa: S113
+        settings.SEE_API_ACCESS_TOKEN_URL, data=payload
+    )
+    response.raise_for_status()
+
+    return response.json()["access_token"]
+
+
+def parse_topics(topic):
+    """
+    Parse topics from a string
+
+    Args:
+        topic (str): the topic string
+
+    Returns:
+        list: the parsed topics
+    """
+
+    return (
+        transform_topics([{"name": topic.split(":")[-1].strip()}], OFFERED_BY_CODE)
+        if topic
+        else []
+    )
+
+
+def parse_image(course_data):
+    """
+    Parse image from course data
+
+    Args:
+        course_data (dict): the course data
+
+    Returns:
+        str: the image URL
+    """
+    image_url = course_data.get("Image_Src")
+    if image_url:
+        return {
+            "url": image_url,
+            "alt": course_data.get("Title"),
+            "description": course_data.get("Title"),
+        }
+    return None
+
+
+def parse_datetime(value):
+    """
+    Parses a datetime string
+
+    Args:
+        value(str): the datetime in string format
+
+    Returns:
+        datetime: the parsed datetime
+    """  # noqa: D401
+    return parse(value).replace(tzinfo=UTC) if value else None
+
+
+def extract():
+    """
+    Extract Sloan Executive Education data
+    """
+    access_token = _get_access_token()
+    courses_response = requests.get(  # noqa: S113
+        urljoin(settings.SEE_API_URL, "courses"),
+        headers={"Authorization": f"JWT {access_token}"},
+    )
+    courses_response.raise_for_status()
+    runs_response = requests.get(  # noqa: S113
+        urljoin(settings.SEE_API_URL, "course-offerings"),
+        headers={"Authorization": f"JWT {access_token}"},
+    )
+    runs_response.raise_for_status()
+
+    courses_data = courses_response.json()
+    runs_data = runs_response.json()
+
+    return courses_data, runs_data
+
+
+def transform_run(run_data, course_data):
+    """
+    Transform Sloan Executive Education run data
+
+    Args:
+        run_data (dict): the run data
+        course_data (dict): the course data
+
+    Returns:
+        dict: the transformed data
+    """
+    return {
+        "run_id": run_data["CO_Title"],
+        "start_date": parse_datetime(run_data["Start_Date"]),
+        "end_date": parse_datetime(run_data["End_Date"]),
+        "title": course_data["Title"],
+        "url": course_data["URL"],
+        "availability": RunAvailability.current.value,
+        "published": True,
+        "prices": [run_data["Price"]],

```

Comment:
I forgot to add instructors, something like this should work:
```python
"instructors": [{"full_name": name.strip()} for name in run_data["Faculty_Name"].split(",")]
```

Reply:
Added instructors to the struct